electrum/gui/qt/channel_details.py

```diff
@@ -143,17 +138,8 @@ def __init__(self, window: 'ElectrumWindow', chan_id: bytes):
         self.format_msat = lambda msat: window.format_amount_and_units(msat / 1000)
         self.format_sat = lambda sat: window.format_amount_and_units(sat)
 
-        # connect signals with slots
-        self.htlc_fulfilled.connect(self.on_htlc_fulfilled)
-        self.htlc_failed.connect(self.on_htlc_failed)
-        self.state_changed.connect(self.do_state_changed)
-        self.htlc_added.connect(self.on_htlc_added)
-
         # register callbacks for updating
-        util.register_callback(self.htlc_fulfilled.emit, ['htlc_fulfilled'])
-        util.register_callback(self.htlc_failed.emit, ['htlc_failed'])
-        util.register_callback(self.htlc_added.emit, ['htlc_added'])
-        util.register_callback(self.state_changed.emit, ['channel'])
+        self.register_callbacks()
```

Comment:
maybe override `def closeEvent(self, event)` and call `unregister_callbacks()` there

Reply:
done in https://github.com/spesmilo/electrum/commit/be322f7ea7e1789fa044e857202cb2a8f24dadde